dao/service.go

```diff
@@ -7,30 +7,42 @@ import (
 
 // Service interface declares how to interact with the persistence layer regardless of underlying technology
 type Service interface {
+
 	// CreateInsolvencyResource will persist a newly created resource
-	CreateInsolvencyResource(dao *models.InsolvencyResourceDao) (int, error)
+	CreateInsolvencyResource(dao *models.InsolvencyResourceDto) (int, error)
 
 	// GetInsolvencyResource will retrieve an Insolvency Resource
 	GetInsolvencyResource(transactionID string) (models.InsolvencyResourceDao, error)
 
 	// CreatePractitionerResourceForInsolvencyCase will persist a newly created practitioner resource in insolvency case
-	CreatePractitionerResourceForInsolvencyCase(dao *models.PractitionerResourceDao, transactionID string) (int, error)
+	CreatePractitionerResourceForInsolvencyCase(transactionID string) (map[string]string, int, error)
 
 	// CreatePractitionerResource will persist a newly created practitioner resource
 	CreatePractitionerResource(dao *models.PractitionerResourceDao, transactionID string) (int, error)
 
-	// GetPractitionerResources will retrieve a list of persisted practitioners
-	GetPractitionerResources(transactionID string) ([]models.PractitionerResourceDao, error)
+	// UpdateInsolvencyPractitioners will update insolvency with practitioners resource
+	UpdateInsolvencyPractitioners(practitionersResource models.InsolvencyResourceDto, transactionID string) (int, error)
+
+	// GetInsolvencyPractitionerByTransactionID will retrieve a list of persisted practitioners
+	GetInsolvencyPractitionerByTransactionID(transactionID string) (string, error)
+
+	// GetPractitionerAppointment will retrieve a practitioner appointment
+	GetPractitionerAppointment(practitionerID string, transactionID string) (*models.AppointmentResourceDao, error)
 
-	// GetPractitionerResource will retrieve a practitioner from the insolvency resource
-	GetPractitionerResource(practitionerID string, transactionID string) (models.PractitionerResourceDao, error)
+	// GetPractitionersByIds will retrieve a practitioner from the insolvency resource
+	GetPractitionersByIds(practitionerIDs []string, transactionID string) ([]models.PractitionerResourceDto, error)
```

Comment:
Is this singular or plural? (comment says 'a practitioner', function name says Practitioners)

Reply:
Fixed